### PR TITLE
Add support for metadata retrieval from GHCR.io

### PR DESCRIPTION
**Pull request by jaimergp**
cc @beckermr

### CODE REVIEW

**Comment by beckermr** [comment]
We'll need to add the OCI mirror package to conda.

**Comment by beckermr** [comment]
You'll also need to change the code to use ruamel.yaml everywhere.

**Comment by jaimergp** [comment]
Ugh I searched for `pyyaml` but of course the import name is just `yaml` 😬 I'll update!

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -34,6 +34,8 @@ jobs:
           mamba install --yes --file=requirements.txt
           mamba install --yes --file=requirements-dev.txt
           python -m pip install -v --no-deps --no-build-isolation -e .
+          # temporary
+          python -m pip install -v https://github.com/channel-mirrors/conda-oci-mirror/archive/018eebd010725eaaf891fbf6cd8438fb1e638b48.tar.gz --no-deps --no-build-isolation
```
Comment (beckermr) [inline .github/workflows/tests.yml:38]:
```suggestion
```

File: pyproject.toml [hunk 2]
```diff
@@ -15,7 +15,8 @@ description = "programatic access to conda-forge's metadata"
 dynamic = ["version"]
 dependencies = [
     "requests",
-    "pyyaml"
+    "ruamel.yaml",
+    "conda-oci-mirror@https://github.com/channel-mirrors/conda-oci-mirror/archive/018eebd010725eaaf891fbf6cd8438fb1e638b48.tar.gz"
```
Comment (beckermr) [inline pyproject.toml:19]: @jaimergp Should we reference a tag here or maybe just main/master?

File: conda_forge_metadata/artifact_info/info_json.py [hunk 3]
```diff
@@ -35,4 +36,9 @@ def get_artifact_info_as_json(channel, subdir, artifact):
             "files": a list of files in the recipe from info/files with
                 elements ending in .pyc or .txt filtered out.
     """
-    return get_libcfgraph_artifact_data(channel, subdir, artifact)
+    if backend == "libcfgraph":
+        return get_libcfgraph_artifact_data(channel, subdir, artifact)
+    elif backend == "oci":
+        return get_oci_artifact_data(channel, subdir, artifact)
```
Comment (beckermr) [inline conda_forge_metadata/artifact_info/info_json.py:42]:
```suggestion
        from conda_forge_metadata.oci import get_oci_artifact_data
        return get_oci_artifact_data(channel, subdir, artifact)
```

File: conda_forge_metadata/artifact_info/info_json.py [hunk 4]
```diff
@@ -1,7 +1,8 @@
 from conda_forge_metadata.libcfgraph import get_libcfgraph_artifact_data
+from conda_forge_metadata.oci import get_oci_artifact_data
```
Comment (beckermr) [inline conda_forge_metadata/artifact_info/info_json.py:2]:
```suggestion
```

File: pyproject.toml (hunk 2 above)
Comment (beckermr) [inline pyproject.toml:19]: Let's move this to an optional dependence.

**Review by beckermr — CHANGES_REQUESTED** [review]
@jaimergp I think for now since conda-oci-mirror is not on pypi and this package ships on both pypi and conda, we should make conda-oci-mirror optional. I have suggested changes to guard the import. All that is left is to make it optional in the pyproject.toml as well.

File: pyproject.toml [hunk 5]
```diff
@@ -15,11 +15,16 @@ description = "programatic access to conda-forge's metadata"
...
     "requests",
-    "pyyaml"
+    "ruamel.yaml"
 ]
 license = {file = "LICENSE"}
 readme = "README.md"
 
+[project.optional-dependencies]
+oci = [
+  "conda-oci-mirror"
```
Comment (beckermr) [inline pyproject.toml:25]: Is this on pypi or do we need to put a git url here?

**Review by beckermr — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 5 above)
Comment (jaimergp) [reply pyproject.toml:25]: We can put a URL too. I'll use the tag URL.

Comment (jaimergp) [reply pyproject.toml]: Let's see now.

**Comment by beckermr** [comment]
Weird test failure but the install appears to work.

**Comment by jaimergp** [comment]
Ugh, that's the pre-commit thingy pyupgrading a bit too much. Let me check.

**Comment by jaimergp** [comment]
There we go!

**Review by beckermr — APPROVED** [review]

**Pull request merged**